abstract/encoding.go

```diff
@@ -170,13 +193,31 @@ func (de *decoder) value(v reflect.Value, depth int) error {
 			}
 		}
 
-	case reflect.Slice, reflect.Array:
+	case reflect.Slice:
+		if v.IsNil() {
+			panic("slices must be initialized to correct length before decoding")
+		}
+		fallthrough
+	case reflect.Array:
 		l := v.Len()
 		for i := 0; i < l; i++ {
 			if err = de.value(v.Index(i), depth+1); err != nil {
 				return err
 			}
 		}
+
+	case reflect.Int:
```

Comment:
Suppose you have a 64-bit server and a 32-client. A structure that holds the number of milliseconds since the first round is integrated as 'int' and sent from the server to the client. Now when testing things on your 64-bit computer, everything is going well. Deploying goes well, the first week you're happy.

Unfortunately, after one month your clients start failing, because now the 32-bit can't hold anymore the number of milliseconds. While debugging you restart the server, and everything is OK again, so you're happy for the next month, before things start going bad again.